config-rules.tf

```diff
@@ -295,6 +299,51 @@ resource "aws_config_config_rule" "required-tags" {
     owner             = "AWS"
     source_identifier = "REQUIRED_TAGS"
   }
+}
+
+
+#
+# The rules that follow have been taken from the following module:
+#  https://github.com/osodevops/aws-terraform-module-config-rules/blob/master/aws_config_rules.tf
+#
+
+data "template_file" "aws_config_ami_approved_tag" {
+  template = "${file("${path.module}/config-policies/ami-approved-tag.tpl")}"
+
+  vars = {
+    ami_required_tag_key_value = var.ami_required_tag_key_value
+  }
+}
+
+#==================================================
+# 0.1	Checks whether running instances are using specified AMIs.
+#==================================================
+resource "aws_config_config_rule" "approved-amis-by-tag" {
+  count             = var.check_approved_amis_by_tag ? 1 : 0
+  name              = "approved-amis-by-tag"
+  description       = "Checks whether running instances are using specified AMIs. Running instances that dont have at least one of the specified tags are noncompliant"
+  input_parameters  = "${data.template_file.aws_config_ami_approved_tag.rendered}"
+
+  source {
+    owner               = "AWS"
+    source_identifier   = "APPROVED_AMIS_BY_TAG"
+  }
+
+  depends_on = [aws_config_configuration_recorder.main]
+}
+
+#==================================================
+# 0.3	Ensure EBS volumes that are in an attached state are encrypted.
+#==================================================
+resource "aws_config_config_rule" "ec2-encrypted-volumes" {
+  count         = var.check_ec2_encrypted_volumes ? 1 : 0
+  name          = "ec2-volumes-must-be-encrypted"
+  description   = "Evaluates whether EBS volumes that are in an attached state are encrypted. Optionally, you can specify the ID of a KMS key to use to encrypt the volume."
+
+  source {
+    owner               = "AWS"
+    source_identifier   = "ENCRYPTED_VOLUMES"
+  }
 
   depends_on = [aws_config_configuration_recorder.main]
```

Comment:
It looks like you've removed `depends_on = [aws_config_configuration_recorder.main]` from the `required-tags` config rule.

Reply:
Great catch here, `depends_on` back in place, sorry about that.

```terraform
resource "aws_config_config_rule" "required-tags" {
  count = var.check_required_tags ? 1 : 0

  name        = "required-tags"
  description = "Checks if resources are deployed with configured tags."

  scope {
    compliance_resource_types = var.required_tags_resource_types
  }

  input_parameters = "${jsonencode(var.required_tags)}"

  source {
    owner             = "AWS"
    source_identifier = "REQUIRED_TAGS"
  }

  depends_on = [aws_config_configuration_recorder.main]
}
```